### ISSUE TITLE
Have slider peak out

### ISSUE THREAD

**Issue by omakasebar** [emilkowalski/vaul#38]
Add a prop or option to have the slider peak into the visible viewport as minimum default height. Clicking the visible slider would bring it to full height, sliding down would slide back down to minimum default height.

![IMG_7810](https://github.com/emilkowalski/vaul/assets/70670294/4dd9072a-b958-467f-844d-b06f2d2aa01c)

**Comment by emilkowalski** [comment]
I've added snap points which makes this possible. You can check out the readme for now. I'll add some codesandbox examples soon!

**Comment by evanwinter** [comment]
Can anyone provide an example of this? Struggling to get it to work a certain way.

I'm looking for a way to make it so the 'closed' state is where it peeks out, and the 'open' state is ~96% height like in most examples.

I tried `snapPoints={[0.2, 1]}` but what I'm seeing is that it opens to 20%, and closes to 0%.

Part of my issue might be that I'm trying to put my Drawer.Trigger inside of Drawer.Content — which might not be possible?

Basically I'm trying to achieve a sticky bottom navigation (a drawer that peeks out 20%), with a button that opens the drawer.

**Comment by emilkowalski** [comment]
Have you seen this example @evanwinter? https://codesandbox.io/p/sandbox/drawer-non-dismissable-forked-jchtff?file=/app/my-drawer.tsx:1,1

**Comment by evanwinter** [comment]
@emilkowalski I did! It does seem to hint at what I'm looking for but I'm having trouble taking it to where I want to go from there.

And to be clear, I'm not sure if what I'm asking about is even possible currently. Having a trigger inside of the content itself seems like it could be a problem.

I'm looking for a way to basically only have two possible states:
1. Open to 20%
2. Open to 95%

I have tried using a combination of `open={true}`, `dismissable={false}`, and snap points, but what I keep seeing is that clicking the trigger when it is inside of the drawer content itself always seems to dismiss the drawer, even with `dismissable={false}`

My use case, in case it helps explain, is a persistent/fixed bottom navigation bar, that the user can tap to expand upwards to reveal the items in their cart.

This is the '20%' state:
![Screenshot 2023-09-18 at 9 53 22 AM](https://github.com/emilkowalski/vaul/assets/19195468/ab820279-4735-4478-8d15-ee29f3b1f256)

And this is the '95%' state:
![Screenshot 2023-09-18 at 9 54 23 AM](https://github.com/emilkowalski/vaul/assets/19195468/27f7c9a9-f14c-482e-b9d4-6fd6bb5e02da)

**Comment by evanwinter** [comment]
Here's a codesandbox of what I'm generally trying to do if that helps too:

https://codesandbox.io/p/sandbox/drawer-snap-points-forked-ccp9cp?file=%2Fapp%2Fmy-drawer.tsx%3A16%2C7-18%2C24

**Comment by siempreeselfindelmundo** [comment]
Hi @evanwinter, were you finally able to make it work? 
i'm currently needing the exact same behaviour on a component and struggling to make it work.

Any help is appreciated, thanks!

**Comment by siempreeselfindelmundo** [comment]
Hi @evanwinter, i found a workaround that in my scenario works just fine. 
I just styled the trigger element to mock a peaked out slider, with an option to add the div that mimics the thingy where you would hold to drag, but it activates the slider on tap.
You can see it here: [https://codesandbox.io/p/sandbox/drawer-snap-points-forked-xr5kj8?file=%2Fapp%2Fmy-drawer.tsx%3A32%2C1](https://codesandbox.io/p/sandbox/drawer-snap-points-forked-xr5kj8?file=%2Fapp%2Fmy-drawer.tsx%3A32%2C1)

**Comment by omakasebar** [comment]
Hi @siempreeselfindelmundo seems like the link is expired. Mind sending another one? Was curious to your approach.
